code/game/machinery/bots/cleanbot.dm

```diff
@@ -271,6 +271,12 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 			armed++
 			icon_state = "roombot_battle[on]"
 			icon_initial = "roombot_battle"
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		var/obj/machinery/bot/cleanbot/roomba/meido/M = new(get_turf(src))
+		M.name = name
```

Comment:
won't this result in all unnamed maidbots being named cleanbot? It's a very minor thing but you could check if the name is equal to the initial name first.

Reply:
tru